android/src/main/java/io/invertase/firebase/links/RNFirebaseLinks.java

```diff
@@ -43,37 +42,61 @@ public String getName() {
   }
 
   @ReactMethod
-  public void getInitialLink(Promise promise) {
-    promise.resolve(initialLink);
+  public void getInitialLink(final Promise promise) {
+    Activity activity = getCurrentActivity();
+    if (activity != null) {
+      FirebaseDynamicLinks.getInstance()
+        .getDynamicLink(activity.getIntent())
+        .addOnSuccessListener(activity, new OnSuccessListener<PendingDynamicLinkData>() {
```

Comment:
Looks like this code is duplicated

Reply:
You mean the builder call? I woudnt say duplicate, but i'll try to prettify it.

Reply:
I mean the resolving of the link:
`FirebaseDynamicLinks.getInstance().getDynamicLink(activity.getIntent())...`

I would imagine that we can just have a function:
`resolveLink(intent, onSuccess, onError)` where `onSuccess` is a handler that gets the `string` of the resolved link (or `null`) and `onError` is called with the `exception`.